### DISCUSSION TITLE
Use pre-commit CI

### DISCUSSION THREAD

**Discussion by foster999** [executablebooks/meta#445] [Ideas] [answered]
I wondered if EBP have considered making use of [pre-commit CI](https://pre-commit.ci/) , rather than running pre-commit checks through Actions?

From my limited use, it offers a couple of additional features:
1. Notification (PR created) when hook repos are updated
2. Automatically commits pre-commit fixes to PRs

Annecdotally, I've found that one-time contributors are less likely to follow contribution guidance, including installing pre-commit hooks. 2 above applies formatting fixes (e.g. 'black') once the PR has been created, reducing the burden on the contributor.

Keen to hear any thoughts 😊

**Comment by chrisjsewell** [comment, marked as answer]
heya, yep you're to slow lol, we've already started adding it to repos

**Comment by choldgraf** [comment]
As @chrisjsewell notes we actually just started using precommit.ci about 2 days ago 😅 agreed it is very useful!

**Reply by foster999** [reply]
Great minds 😉 is it on for every repo?

I mostly work on `sphinx-tabs`, where this would come in handy.

**Reply by chrisjsewell** [reply]
Yeh, it will just be rolled on a gradual basis really, as obviously it's not that urgent.
I have just turned it on for sphinx-tabs

**Reply by foster999** [reply]
Thanks, that's great 😊
